api/metadata_cache.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"google.golang.org/appengine/memcache"
+)
+
+const metadataCacheKey = "WPT-METADATA"
+
+type webappMetadataFetcher struct {
+	ctx    context.Context
+	client *http.Client
+	log    shared.Logger
```

Comment:
Optional: we don't really need to have this field. `Fetch` can simply get a logger from `ctx` and pass it to `getMetadataFromMemcache`. This also works in tests: `TestContext` has a nil logger associated with it already.